### PR TITLE
Move metadata join upstream

### LINKED ISSUE #214: When should we join the VDS raw data with the metadata?
We've gone back and forth on this a few times with @kengodleskidot in the context of some of our large aggregations on 30-second VDS data. Some of our downstream metrics depend on detector metadata, which lives in the `STATION_META` and related tables, and not on the raw tables. In order to compute these metrics, we need to join the two tables together on `STATION_ID`.

Normally, we'd do that join as late as possible, so that filtering and especially aggregation can occur, allowing for a cheaper join. For instance, if we join on the 5-minute data, that's 10x smaller of a join than the 30 second data, so we would want to do that if we don't need any of the metadata to do the 5-minute aggregation.

However, a quirk of our base tables is that they are delivered in wide format, with `FLOW_1`, `FLOW_2`, `FLOW_3` ...`FLOW_8`, etc. But not every station has eight lanes of traffic! In fact, the modal one has two or three. So when we do some of our large aggregations, we are doing them on more lanes than exist, wasting a lot of effort.

Right now we are doing the aggregations first, and joining with the metadata second. But we should experiment with joining at other stages! We'd be paying higher costs for the joins, but lower costs on the aggregations.

Example DAGs showing the late join with the metadata:

image

image

### PR DESCRIPTION
**Pull request by JamesSLogan**
Changes:
- Added metadata to 5 minute aggregation models (this gives us `detector_id`)
- Removed downstream metadata joins
- Disambiguated `id` by replacing with `station_id` 
- Updated some models to key off of `detector_id` instead of `station_id` and `lane`

Fixes #214

### CODE REVIEW

File: transform/models/intermediate/clearinghouse/int_clearinghouse__station_agg_five_minutes.sql [hunk 1]
```diff
@@ -27,22 +27,28 @@ joined_data as (
...
         sample_date,
         sample_timestamp,
-        district,
+        any_value(freeway) as freeway,
+        any_value(direction) as direction,
+        any_value(station_type) as station_type,
+        any_value(absolute_postmile) as absolute_postmile,
+        any_value(district) as district,
+        any_value(latitude) as latitude,
+        any_value(longitude) as longitude,
```
Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/int_clearinghouse__station_agg_five_minutes.sql:45]: Love this

File: transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql [hunk 2]
```diff
@@ -52,7 +52,7 @@ monthly_station_level_spatial_temporal_metrics as (
 
         {% endfor %}
     from station_daily_data
-    group by id, sample_month, city, county, district, type, freeway, direction, length
+    group by station_id, sample_month, length
```
Comment (ian-r-rose) [inline transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql:55]: Here and above: is there a reason you haven't moved `length` into the `any_value` block?

File: transform/models/intermediate/clearinghouse/int_clearinghouse__detector_agg_five_minutes.sql [hunk 3]
```diff
@@ -1,13 +1,20 @@
 {{ config(
     materialized="incremental",
     cluster_by=["sample_date"],
-    unique_key=["id", "lane", "sample_timestamp"],
+    unique_key=["detector_id", "sample_timestamp"],
     snowflake_warehouse = get_snowflake_refresh_warehouse(small="XL")
```
Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/int_clearinghouse__detector_agg_five_minutes.sql:5]: I'm curious if you have done any testing on how much longer this table takes to build. Can we get away with bringing the warehouse sizes down?

File: transform/models/intermediate/imputation/int_imputation__detector_agg_five_minutes.sql [hunk 4]
```diff
@@ -13,12 +13,6 @@ with base as (
     where {{ make_model_incremental('sample_date') }}
 ),
 
-/* Configuration data for detectors, filtered to mainline or HOV, as the main travel lanes */
-detector_config as (
-    select * from {{ ref("int_vds__detector_config") }}
-    where station_type in ('ML', 'HV') -- TODO: make a variable for "travel station types"
-),
-
```
Comment (ian-r-rose) [inline transform/models/intermediate/imputation/int_imputation__detector_agg_five_minutes.sql:21]: 🎉

**Review by ian-r-rose — COMMENTED** [review]
Thanks @JamesSLogan this is really great work. I see there are a couple of small merge conflicts, and we may have to go through a couple of contortions to get CI passing with the incremental models, but overall I think this is looking basically ready to go in.

I'm curious if you did any performance testing to see how expensive this join is at the end of the day?

File: transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql (hunk 2 above)
Comment (JamesSLogan) [reply transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql:55]: length is used in the delay calculations, as it turns out. It's probably possible to rework things to get around this but I largely stayed away from major logic changes.

Comment (ian-r-rose) [reply transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql:55]: I'm not sure I understand -- you've moved the other group-by keys into the list of `any_value`s, how is length different, regardless of how it's used downstream?

Comment (JamesSLogan) [reply transform/models/intermediate/performance/int_performance__station_metrics_agg_monthly.sql]: Fixed. Using `any_value(length) as length ` previously gave an error, but modifying the delay calculation to use the aggregated length fixes this.

**Review by ian-r-rose — APPROVED** [review]
Nice work @JamesSLogan!

**Pull request merged**